lib/boku/client.py

```diff
@@ -282,6 +290,20 @@ def check_transaction(self, transaction_id):
             'paid': Decimal(tree.find('paid').text),
         }
 
+    def check_sig(self, data):
+        """
+        Check that a signature is valid. If solitude is run with the proxy,
+        this check has to be done on the proxy, because the solitude database
+        does not have access to the required BOKU_SECRET_KEY.
+
+        This takes the output of `get_pairs_string` as data.
+
+        Returns the value of the signature calculated using BOKU_SECRET_KEY.
+        """
```

Comment:
Because of your comment about how this needs to run on the proxy I wonder if it should raise an error if ``settings.BOKU_SECRET_KEY`` is empty. Otherwise, it will silently create the wrong signature (probably using an empty string as the secret)